library/core/src/ffi/c_str.rs

```diff
@@ -296,6 +299,31 @@ impl CStr {
     #[must_use]
     #[stable(feature = "rust1", since = "1.0.0")]
     #[rustc_const_stable(feature = "const_cstr_from_ptr", since = "1.81.0")]
+    // Preconditions:
+    // - ptr must be non-null and properly aligned
+    // - ptr must point to a valid nul-terminated string within isize::MAX bytes
```

Comment:
I guess we are not checking that, eh? You could check that the distance between next and sub is smaller than `isize::MAX`.